aws/sdk/integration-tests/s3/tests/express.rs

```diff
@@ -353,3 +353,34 @@ async fn disable_s3_express_session_auth_at_operation_level() {
         "x-amz-create-session-mode should not appear in headers when S3 Express session auth is disabled"
     );
 }
+
+#[tokio::test]
+async fn support_customer_overriding_express_credentials_provider() {
+    let (http_client, rx) = capture_request(None);
+    let expected_session_token = "testsessiontoken";
+    let client = test_client(|b| {
+        b.http_client(http_client.clone())
+            // Pass a credential with a session token so that
+            // `x-amz-s3session-token` should appear in the request header.
+            .express_credentials_provider(Credentials::new(
+                "testaccess",
+                "testsecret",
+                Some(expected_session_token.to_owned()),
+                None,
+                "test",
+            ))
+    })
+    .await;
+    let _ = client
+        .list_objects_v2()
+        .bucket("s3express-test-bucket--usw2-az1--x-s3")
+        .send()
+        .await;
+
+    let req = rx.expect_request();
+    let actual_session_token = req
+        .headers()
+        .get("x-amz-s3session-token")
+        .expect("x-amz-s3session-token should be present");
+    assert_eq!(expected_session_token, actual_session_token);
```

Comment:
consider adding a test that hits a non-express bucket and ensuring that this token _is not_ used

Reply:
Added in 574b97a